tests/components/pegel_online/test_sensor.py

```diff
@@ -1,53 +1,141 @@
 from homeassistant.core import HomeAssistant
 
 from . import PegelOnlineMock
+from .const import (
+    MOCK_CONFIG_ENTRY_DATA_DRESDEN,
+    MOCK_CONFIG_ENTRY_DATA_HANAU_BRIDGE,
+    MOCK_CONFIG_ENTRY_DATA_WUERZBURG,
+    MOCK_STATION_DETAILS_DRESDEN,
+    MOCK_STATION_DETAILS_HANAU_BRIDGE,
+    MOCK_STATION_DETAILS_WUERZBURG,
+    MOCK_STATION_MEASUREMENT_DRESDEN,
+    MOCK_STATION_MEASUREMENT_HANAU_BRIDGE,
+    MOCK_STATION_MEASUREMENT_WUERZBURG,
+)
 
 from tests.common import MockConfigEntry
 
-MOCK_CONFIG_ENTRY_DATA = {CONF_STATION: "3bcd61da-xxxx-xxxx-xxxx-19d5523a7ae8"}
-
-MOCK_STATION_DETAILS = Station(
-    {
-        "uuid": "3bcd61da-xxxx-xxxx-xxxx-19d5523a7ae8",
-        "number": "501060",
-        "shortname": "DRESDEN",
-        "longname": "DRESDEN",
-        "km": 55.63,
-        "agency": "STANDORT DRESDEN",
-        "longitude": 13.738831783620384,
-        "latitude": 51.054459765598125,
-        "water": {"shortname": "ELBE", "longname": "ELBE"},
-    }
-)
-MOCK_STATION_MEASUREMENT = CurrentMeasurement("cm", 56)
-
 
-async def test_sensor(hass: HomeAssistant) -> None:
+@pytest.mark.parametrize(
+    (
+        "mock_config_entry_data",
+        "mock_station_details",
+        "mock_station_measurement",
+        "expected_states",
+    ),
+    [
+        (
+            MOCK_CONFIG_ENTRY_DATA_DRESDEN,
+            MOCK_STATION_DETAILS_DRESDEN,
+            MOCK_STATION_MEASUREMENT_DRESDEN,
+            {
+                "sensor.dresden_elbe_water_volume_flow": (
+                    "DRESDEN ELBE Water volume flow",
+                    "88.4",
+                    "m³/s",
+                ),
+                "sensor.dresden_elbe_water_level": (
+                    "DRESDEN ELBE Water level",
+                    "62",
+                    "cm",
+                ),
+            },
+        ),
+        (
+            MOCK_CONFIG_ENTRY_DATA_HANAU_BRIDGE,
+            MOCK_STATION_DETAILS_HANAU_BRIDGE,
+            MOCK_STATION_MEASUREMENT_HANAU_BRIDGE,
+            {
+                "sensor.hanau_brucke_dfh_main_clearance_height": (
+                    "HANAU BRÜCKE DFH MAIN Clearance height",
+                    "715",
+                    "cm",
+                ),
+            },
+        ),
+        (
+            MOCK_CONFIG_ENTRY_DATA_WUERZBURG,
+            MOCK_STATION_DETAILS_WUERZBURG,
+            MOCK_STATION_MEASUREMENT_WUERZBURG,
+            {
+                "sensor.wurzburg_main_air_temperature": (
+                    "WÜRZBURG MAIN Air temperature",
+                    "21.2",
+                    "°C",
+                ),
+                "sensor.wurzburg_main_oxygen_level": (
+                    "WÜRZBURG MAIN Oxygen level",
+                    "8.4",
+                    "mg/l",
+                ),
+                "sensor.wurzburg_main_ph_value": (
+                    "WÜRZBURG MAIN pH value",
+                    "8.1",
+                    "--",
```

Comment:
pH should have no unit, not `--`
Do you need to have a conversion betwen these and HA UoM's

Reply:
> pH should have no unit, not `--`

unfortunately the uom come from the api

> Do you need to have a conversion betwen these and HA UoM's

would be great to have, but not sure if mandatory is this case.
i guess you are thinking about, to use the unit string enums?

Reply:
i've added device classes where applicable. for ph-value entity the uom is overwritten by `None`

Reply:
I was more thinking that you could have a dict containing the "translation"
```
API_TO_HA_UOMS = {
    "--": None
    ....
}
and then use that
`self._attr_native_unit_of_measurement = API_TO_HA_UOMS[self.measurement.uom]`

Maybe only pH is different so it's not so much needed.

Reply:
as far as i can see, only the ph has such uggly uom, some others uses uom which are not covered by any of these `UnitOfXXX` enums (_eq. `m³/s` from the water volume flow_)